### PR TITLE
Add Mangazure

### PR DESCRIPTION
**Pull request by MikeZeDev**
https://github.com/manga-download/hakuneko/issues/3376

### CODE REVIEW

File: web/src/engine/websites/Mangazure.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            for (let offset = 1, run = true; run; offset += perPage) {
+                const url = new URL(`/category?q=${category}&start=${offset}&max=${perPage}`, this.apiUrl);
+                const request = new Request(url, {
+                    headers: {
+                        Referer: this.URI.href,
+                        Origin: this.URI.href
+                    }
+                });
+                const results = await FetchJSON<APIitem[]>(request);
+                results.forEach(manga => mangaList.push(new Manga(this, provider, new URL(manga.url).pathname, this.cleanMangaTitle(manga.title))));
```
Comment (ronny1982) [inline web/src/engine/websites/Mangazure.ts:57]: The identifier of the manga seems to be incorrect, identifier should be `this.convertToSlug(manga.Title)`

File: web/src/engine/websites/Mangazure.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        return mangaList.distinct();
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const slug = this.convertToSlug(manga.Title);
+        const url = new URL(`/book?q=${slug}`, this.apiUrl);
+        const request = new Request(url, {
+            headers: {
+                Referer: this.URI.href,
+                Origin: this.URI.href
```
Comment (ronny1982) [inline web/src/engine/websites/Mangazure.ts:70]: The origin header must not include path, search or fragments, use `this.URI.origin` instead.

File: web/src/engine/websites/Mangazure.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const response = await FetchWindowScript<PageData>(new Request(new URL(chapter.Identifier, this.URI)), pageScript, 2000);
+        const pages = response.rawdata.match(/\$chapterContent\s*=\s*"([^"]+)/)[1].replace(/<.*>/, '').split('|');
+        return pages
+            .filter(page => page != '')
+            .map(page => new Page(this, chapter, new URL(this.getImageUrl(page, response.server))));
+    }
+
+    getImageUrl(path: string, server: string): string {
```
Comment (ronny1982) [inline web/src/engine/websites/Mangazure.ts:85]: Suggestion to place this into the page script.

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/Mangazure.ts (hunk 3 above)
Comment (MikeZeDev) [reply web/src/engine/websites/Mangazure.ts:85]: meh, script doesnt like regexp syntax for some reason, i would like to avoid headeaches but ill try

Comment (MikeZeDev) [reply web/src/engine/websites/Mangazure.ts:85]: For some reasons doing the regexp inside the script returns nothing 🤷

**Pull request merged**